### PR TITLE
remove "kind:" from examples and docs where optional

### PR DESCRIPTION
**Pull request by kkr16**
### Description

- updated [examples/README.md](https://github.com/GoogleCloudPlatform/hpc-toolkit/compare/develop...kkr16:develop?expand=1#diff-49aaa2819e35a856818ecec8c9fa7e1c79ad028d3f44bd749736353cfb51bac9) to state that `kind` is optional and default is `terraform`
- removed `kind: terraform` from docs and examples 

### Submission Checklist

* [x] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [x] Are all tests passing? (`make tests`)
* [x] Have you written unit tests to cover this change? 
* [x] Is unit test coverage still above 80%?
* [x] Have you updated all applicable documentation?
* [x] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

**Review by heyealex — CHANGES_REQUESTED** [review]
Thank you for the PR submission! This change touched a lot of files, and I appreciate your attention to detail. 

I just have the one comment I'd like to have addressed, and that should just be a `git restore` on a one or two of the test_configs to ensure the optional variable is still being validated in `make tests`.

File: tools/validate_configs/test_configs/2-nfs-servers.yaml [hunk 1]
```diff
@@ -27,11 +27,9 @@ deployment_groups:
   modules:
   - id: network1
     source: modules/network/pre-existing-vpc
-    kind: terraform
```
[the remark this answers is not in the document]
Comment (kkr16) [reply tools/validate_configs/test_configs/2-nfs-servers.yaml:30]: I restored two tests to include `kind: terraform`
- `tools/validate_configs/test_configs/packer.yaml`
- `tools/validate_configs/test_configs/threads_per_core.yaml`

**Comment by kkr16** [comment]
> Thank you for the PR submission! This change touched a lot of files, and I appreciate your attention to detail.
> 
> I just have the one comment I'd like to have addressed, and that should just be a `git restore` on a one or two of the test_configs to ensure the optional variable is still being validated in `make tests`.

Thank you for the feedback! 
Comment addressed. ptal.

**Review by heyealex — APPROVED** [review]
Thanks for the update, this looks good!

**Pull request merged**